homeassistant/components/elmax/config_flow.py

```diff
@@ -170,82 +177,65 @@ async def async_step_panels(
 
     async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
         """Perform reauth upon an API authentication error."""
-        self._reauth_username = entry_data.get(CONF_ELMAX_USERNAME)
-        self._reauth_panelid = entry_data.get(CONF_ELMAX_PANEL_ID)
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])  # type: ignore[assignment]
         return await self.async_step_reauth_confirm()
 
-    async def async_step_reauth_confirm(self, user_input=None):
+    async def async_step_reauth_confirm(self, user_input=None) -> FlowResult:
         """Handle reauthorization flow."""
         errors = {}
         if user_input is not None:
-            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
-            password = user_input.get(CONF_ELMAX_PASSWORD)
-            entry = await self.async_set_unique_id(self._reauth_panelid)
+            username = user_input[CONF_ELMAX_USERNAME]
+            password = user_input[CONF_ELMAX_PASSWORD]
+            panel_pin = user_input[CONF_ELMAX_PANEL_PIN]
 
             # Handle authentication, make sure the panel we are re-authenticating against is listed among results
             # and verify its pin is correct.
+            reauth_panel_id = self._entry.data[CONF_ELMAX_PANEL_ID]
```

Comment:
```suggestion
            assert self._entry is not None
            reauth_panel_id = self._entry.data[CONF_ELMAX_PANEL_ID]
```

Reply:
Done in 59b5c7f6e6cbebd005e9b74b73da31fa5612cb94, thank you!